gherkin/java/src/main/java/io/cucumber/gherkin/GherkinDialect.java

```diff
@@ -60,6 +64,18 @@ public List<String> getStepKeywords() {
         return result;
     }
 
+    public Map<StepKeywordType, List<String>> getStepKeywordsByType() {
```

Comment:
I think it would be most useful to return `Map<String, StepKeywordType>` here, as it would make the lookup in TokenMatcher simpler

Or possibly it would make more sense to have `public StepKeywordType getKeywordType(String keyword)` here?

Reply:
Thinking about your second suggestion, I think that'll allow me to restore the original iteration over all keywords as it was originally coded, retrieving the keyword type through this function call. However, the function itself would need to do roughly the same as the inner loop right now (or it can do this in advance when selecting the dialect): It'll need to determine whether a keyword translation associates with more than one keyword (Given/When/Then/But/And).

Reply:
Implemented something along these lines and it simplifies `match_StepLine()` indeed.